blog: Add October 2025 project update (#32)
AerynOS/dotcom: src/content/docs/blog/2025/10/31/index.mdx

---
title: "October 2025 project update"
date: 2025-10-31T09:00:00Z
authors: [nomadiccore, ermo]
tags: [news]
---

import Kofi from "@/components/ui/Kofi.astro";

We are firmly in the final quarter of 2025 and what a year it's been so far. In our last blog post at the end of September, we mentioned that we were delaying the release of our next ISO into October to give our Gnome 49 stack (and the wider extensions ecosystem) more time to mature.

Coming into the final week of October, the team made the decision to transition from clang's libc++ to GNU libstdc++ and work through the associated rebuild requirements across our repository.

We also mentioned in our September blog post that we had reached the point of having stable build infrastructure. However, over the course of October, we had an old problem re-appear, which proved somewhat vexing to solve initially.

After the issue was debugged, and a fixed version of boulder was deployed to the build infrastructure, we used the transition back to libstdc++, which included hundreds and hundreds of rebuilds, and the landing of KDE Plasma 6.5 stack to verify that that we have put this particular issue behind us.

> In addition to the boulder threading bug fixes, the team has also found the time to hash out a design for, and prepare an early prototype PR of, the moss declarative `system-model` feature, as well as landing a few small user experience improvements and correctness fixes to both moss and boulder.

Reviewer: ```suggestion
In addition to the build infrastructure related boulder fixes, the team has also found the time to hash out a design for, and prepare an early prototype PR of, the moss declarative `system-model` feature, as well as landing a few small user experience improvements and correctness fixes to both moss and boulder.
```

Additional details about Cosmic can be found on System76's [website](https://system76.com/cosmic).

### Gnome

The Gnome packaging team has updated our Gnome environment to 49.1. The team has also expanded the number of available Gnome packages in our repository, and we now include Showtime and Gnome-contacts in our gnome pkgsets accordingly. For clarity, these were added in September but were not mentioned in our previous blog post.

Gnome continues to work very well on AerynOS and remains as our default live installation environment.

### KDE Plasma

Reilly Brogan has done a fantastic job landing KDE Plasma 6.5, KDE Gear 25.08.2 and KDE Frameworks 6.19.0 into our repository over the last month. The overall KDE Plasma experience continues to improve with users providing largely very positive feedback on its performance and fluidity on AerynOS.

Over the last couple of months, KDE Plasma has now also been promoted as a recommended installation option alongside Gnome.

## Distribution Updates

### Switching back to GNU libstdc++ from LLVM libcxx

> The team decided to switch back to GNU libstdc++ from the LLVM libcxx C++ library as a defensive measure. This has resulted in us having to carry fewer patches across the stack, and resolved a few bugs as bonus. In particular, this has squashed an annoying bug related to the Firefox Widevine DRM plugin that in turn previously made certain video conferencing tools crash.

Reviewer: ```suggestion
The team decided to switch back to the GNU libstdc++ library from the LLVM libcxx C++ library as a defensive measure. This has resulted in us having to carry fewer patches across the stack, and has resolved a few bugs as a bonus. In particular, this has squashed an annoying bug related to the Firefox Widevine DRM plugin that in turn previously made certain video conferencing tools crash.
```

### Packaging License Matching

> New this month, when packagers use the "boulder recipe new" invocation to create a new recipe from an upstream source archive, boulder will attempt to identify the applicable license of the package, and add it to the autogenerated skeleton stone.yaml recipe file.

Reviewer: ```suggestion
New this month, when packagers use the `boulder recipe new` invocation to create a new recipe from an upstream source archive, boulder will attempt to identify the applicable license of the package, and add it to the autogenerated skeleton stone.yaml recipe file.
```

However, as these issues only manifested for us when boulder was invoked by the build infrastructure, and even then only rarely, it was not a problem we could trigger on demand.

After some head-scratching and debugging sessions where we attached debuggers to live builds that had happened to hang, we finally found the root cause, and in turn were able to simplify our threading code significantly by always using a dedicated, explicit runtime that gets automatically shut down once the parallel (rayon) or threaded (tokio) work unit goes out of scope.

This in turn now ensures that all threads have been shut down as required, before entering the cloned user-namespace container.

Massive thanks to Cory Forsstrom for the work he put into solving this problem very elegantly over a couple of Pull Requests.

### moss `system-model` prototype feature

> The system-model feature is inspired by the feature of the same name from the Conary system and package manager. However, in practice, it will likely end up more like a hybrid of the Conary feature and e.g. the simpler and more limited Gentoo (and FreeBSD) 'world' definition.

Reviewer: ```suggestion
The system-model feature is inspired by the feature of the same name from the Conary system and package manager. However, in practice, it will likely end up more like a hybrid of the Conary feature and e.g. the simpler and more limited Gentoo (and FreeBSD) 'world' set.
```

We are still hoping to be able to, when the time comes, offer versioned system-models that match our planned versioned repository work.

> The goal of the moss system-model feature is that it will eventually enable us to define, and switch between, system installs declaratively and seamlessly. This feature is slated to be suitable for use on both on live systems, recovery initramfs environments, and for installing new systems.

Reviewer: ```suggestion
The goal of the moss system-model feature is that it will eventually enable us to define, and switch between, system installs declaratively and seamlessly. This feature is slated to be suitable for use on both live systems, recovery initramfs environments, and for installing new systems.
```

We hope that, as we evolve and flesh out the system-model feature work, this will give both users and system integrators the flexibility they need to define their systems in ways that matches their requirements and preferences.

## Wider Project updates

### Donations

During October, we have made changes to our donation accounts and updated our site accordingly. Most importantly, the primary way to now provide donations to AerynOS is via our Ko-fi [page](https://ko-fi.com/aerynos).

Due to the way Ko-Fi works, we have had to cancel all existing recurring donations as the money would not automatically flow to our new accounts. We have sent messages out to our donors and are hoping that they will want to sign-up again to provide continued donations towards the project.

All donations received by AerynOS are going towards paying our running costs, reimbursing ermo for his project hardware investments, and building a buffer for acquiring new hardware once the current hardware is EOL.

Taken together, these three budget items result in a target income of around €500 per month, with a third of that projected to go to each of the outlined items.

> Due to the cancellations mentioned above, we are currently down to €25 of new/commited monthly recurring donations.

Reviewer: ```suggestion
Due to the cancellations mentioned above, we are currently down to €25 of new/committed monthly recurring donations.
```
